src/cmds/asset.js

```diff
@@ -2,31 +2,77 @@ const path = require('path')
 const fs = require('fs')
 const { default: fetch } = require('node-fetch')
 
+const mediasUrl = 'https://plugins.livestorm.co/api/v1/medias'
+let directory = {}
+
 module.exports = function asset() {
-  let file = process.argv[3]
-  if (!file) {
+  let givenPath = process.argv[3]
+  if (!givenPath) {
     console.log('Please specify the asset to upload')
-    console.log('Usage: livestorm asset <file>')
+    console.log('Usage: livestorm asset <file|directory>')
     process.exit(1)
   }
-  file = path.join(process.cwd(), file)
+  givenPath = path.join(process.cwd(), givenPath)
 
-  if (fs.statSync(file).size / (1024*1024) > 8) {
-    console.log('File size must be inferior to 8MB.')
-    process.exit(1)
+  uploadFileOrDirectory(givenPath)
+}
+
+function uploadFileOrDirectory(givenPath) {
```

Comment:
Il y a beaucoup de fonctions qui commencent par `upload` met qui n'uploade pas vraiement et servent à préparer les véritables uploads. Je pense qu'il faut les distinguer en les nommant différement.

Par exemple ici : `beforeUpload` ou `prepareUpload`

Reply:
Le sens n'est pas vraiment de préparer l'upload je trouve, pour moi le but est de conserver ça le plus haut niveau possible à la lecture afin d'abstraire tout complexité, là en l'occurence l'enchainement part vraiment d'une fonction simpliste à laquelle tu donnes un dossier ou un fichier :
- uploadFileOrDirectory(path)
  - Si c'est un dossier -> uploadEachFileFrom(path)
    - puis de manière récursive -> Pour chaque element du dossier -> uploadFileOrDirectory(element) 🔁
  - Sinon -> uploadFile(path)

Et c'est tout du coup, ça me semble lisible et sémantiquement plus éloquent que "prepareUpload" ou "beforeUpload" qui n'indiquent pas l'intention. Qu'en penses-tu ?

Reply:
Je n'ai sans doute pas indiqué les meilleures propositions. Je pense qu'il aurait été plus optimal de nommer les fonctions `uploadSomething` là où l'upload s'effectue réellement car avant cela, les fonctions servent à déterminer le type d'upload. Mais ce n'est pas bloquant pour moi.

Reply:
D'acc, perso ça ne me dérange pas, in fine l'action faite quand j'appelle uploadFileOrDirectory est bien d'upload l'élement même si certains step sont nécessaire, d'ailleurs si jamais tu sais si ce que tu upload est un dossier ou un fichier tu peux même call directement `uploadFile` ou `uploadEachFileFrom(directory)`